flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/sink/NoopCommitterHandler.java

```diff
@@ -34,7 +34,8 @@
 
     @Override
     public List<Object> processCommittables(
-            SupplierWithException<List<Object>, Exception> committableSupplier) {
+            SupplierWithException<List<Object>, Exception> committableSupplier) throws Exception {
+        committableSupplier.get();
```

Comment:
Why is this needed, what's not working in this way? It feels counter-intuitive to have `preCommit` without `commit`. It seems like I missed a case when I explicitly structured the code to not call `preCommit`.

Reply:
If we always want to `preCommit`, then please refactor such that `preCommit` is called on call-site (e.g. remove the `Supplier` and directly pass the result.

Reply:
> Why is this needed, what's not working in this way? It feels counter-intuitive to have `preCommit` without `commit`. It seems like I missed a case when I explicitly structured the code to not call `preCommit`.

AFAICT all async sinks which want to implement at-least-once cannot work currently. They are all built on the assumption that during `preCommit` `mailbox.yield()` is called until all outstanding message callbacks are called but they do not have any transaction-like to commit (no committer). I personally experienced the issue when trying to implement the elasticseach sink.

> If we always want to `preCommit`, then please refactor such that `preCommit` is called on call-site (e.g. remove the `Supplier` and directly pass the result.

Sure can do that 👍

Reply:
We should adjust documentation to point out that `prepareCommit = commit` on 1-pc sinks. Except from that, the PR is fine.